### DISCUSSION TITLE
[Help]: Some setting page don't load

### DISCUSSION THREAD

**Discussion by Alfonso0p6** [mbentley/docker-omada-controller#394] [Help] [answered]
### Controller Version

5.13.22

### Describe the Bug

some settings pages do not load

### Expected Behavior

pages loading correctly

### Steps to Reproduce

1. Start the container,
2. Login
3. Open the settings

### How You're Launching the Container

```plain
docker run -d \
  --name omada-controller \
  --restart unless-stopped \
  --ulimit nofile=4096:8192 \
  -p 8088:8088 \
  -p 8043:8043 \
  -p 8843:8843 \
  -p 27001:27001/udp \
  -p 29810:29810/udp \
  -p 29811-29816:29811-29816 \
  -e MANAGE_HTTP_PORT=8088 \
  -e MANAGE_HTTPS_PORT=8043 \
  -e PGID="508" \
  -e PORTAL_HTTP_PORT=8088 \
  -e PORTAL_HTTPS_PORT=8843 \
  -e PORT_ADOPT_V1=29812 \
  -e PORT_APP_DISCOVERY=27001 \
  -e PORT_DISCOVERY=29810 \
  -e PORT_MANAGER_V1=29811 \
  -e PORT_MANAGER_V2=29814 \
  -e PORT_TRANSFER_V2=29815 \
  -e PORT_RTTY=29816 \
  -e PORT_UPGRADE_V1=29813 \
  -e PUID="508" \
  -e SHOW_SERVER_LOGS=true \
  -e SHOW_MONGODB_LOGS=false \
  -e SSL_CERT_NAME="tls.crt" \
  -e SSL_KEY_NAME="tls.key" \
  -e TZ=Etc/UTC \
  -v omada-data:/opt/tplink/EAPController/data \
  -v omada-logs:/opt/tplink/EAPController/logs \
  mbentley/omada-controller:5.13
```

### Container Logs

```plain
logs attached
```

### MongoDB Logs

```plain
logs attached
```

### Additional Context

[output.log](https://github.com/mbentley/docker-omada-controller/files/13851683/output.log)
[mongod.log](https://github.com/mbentley/docker-omada-controller/files/13851686/mongod.log)
![image](https://github.com/mbentley/docker-omada-controller/assets/107082461/66fb4e73-dcae-4ceb-b0a3-1940969d6e4b)
![image](https://github.com/mbentley/docker-omada-controller/assets/107082461/6eeda502-1a31-46a0-b6bc-30cc6941d45e)
![image](https://github.com/mbentley/docker-omada-controller/assets/107082461/0da0a630-474f-4b3b-b93b-d7ff2ef21972)
![image](https://github.com/mbentley/docker-omada-controller/assets/107082461/0e3f2652-1924-470c-a188-804310984789)

**Comment by mbentley** [comment]
I can only imagine that this is a browser issue - these pages load fine here.  Check with your browser's dev tools to see if there is some reason why pages aren't loading (frontend errors, API backend errors, etc), try another browser, or clear your cache.

**Comment by Alfonso0p6** [comment]
![image](https://github.com/mbentley/docker-omada-controller/assets/107082461/4b52a0ea-de68-4409-837d-f3be174c9951)
i have this cloudflare tunnel setuped.
![image](https://github.com/mbentley/docker-omada-controller/assets/107082461/068b5386-a4ab-4777-938e-526c62605e60)
this is the browser console

**Reply by mbentley** [reply]
It could be a few things.

Not sure if the cloudflare tunnel is acting as a load balancer/ingress in front of your controller but maybe web sockets aren't working through it.  Can you bypass the cloudflare tunnel in place to check?  Might want to check the dev tools network tab and see if your browser has a "websocket" or "ws" filter to just show the websocket traffic to make sure it isn't returning an error.  You should see a `101` http response like:

![screenshot 2024-01-07 at 10 25 56 AM](https://github.com/mbentley/docker-omada-controller/assets/414445/eb997632-98ef-4b0c-936b-f1631534e48d)

Another thing could just be a browser extension blocking some of the javascript from loading but I am leaning toward the first issue.

**Comment by Alfonso0p6** [comment, marked as answer]
sorry for bothering you, u were right it's cloudflare that in some way is blocking websocket, accessing from the ip works fine
![image](https://github.com/mbentley/docker-omada-controller/assets/107082461/dc8934ea-b59c-481a-9341-f8bb68c507f9)
